File: java/over.java
import java.util.Stack;

class over extends Instruccion{

    @Override
    public void ejecutar(Stack<Integer> pila) {
        int valor1 = pila.pop();
        int valor2 = pila.pop();
        pila.push(valor2);//apilo el segundo
        pila.push(valor1); //apilo el que estaba en la cima
        pila.push(valor2);//apilo de nuevo el segundo
    }

    @Override
    public int incrementarPc(int pc) {
        return pc + 1;
    }

    @Override
    public String mostrar() {
        return " over";
    }
    
}

File: c++/over.cc
#include "over.h"
#include <iostream>

using namespace std;

void over::ejecutar(stack<int>& pila){
    int valor = pila.top();
    pila.pop();
    int valor2 = pila.top();
    pila.pop();//2 valores desapilados
    pila.push(valor2);//apilo 2º desapilado
    pila.push(valor);//apilo 1º desapilado
    pila.push(valor2);//apilo 2º desapilado de nuevo

}

int over::modificar(int pc){
    return pc + 1;
}

std::string over::mostrar() {
    return "    over" ;
}